commons/src/main/java/fr/insideapp/sonarqube/apple/commons/issues/RepositoryRule.java

```diff
@@ -17,45 +17,41 @@
  */
 package fr.insideapp.sonarqube.apple.commons.issues;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+
 public final class RepositoryRule {
 
-    private final String key;
-    private final String name;
-    private final String severity;
-    private final String description;
-    private final String type;
-    private final RepositoryRuleDebt debt;
-
-    public RepositoryRule(final String key, final String name, final String severity, final String description, final String type, final RepositoryRuleDebt debt) {
-        this.key = key;
-        this.name = name;
-        this.severity = severity;
-        this.description = description;
-        this.type = type;
-        this.debt = debt;
+    public enum Severity {
+        BLOCKER,
+        CRITICAL,
+        MAJOR,
+        MINOR,
+        INFO
     }
 
-    public String getKey() {
-        return key;
+    public enum Type {
+        CODE_SMELL,
+        BUG,
+        VULNERABILITY,
+        SECURITY_HOTSPOT
     }
 
-    public String getName() {
-        return name;
-    }
+    @JsonProperty("key")
```

Comment:
So no more read only getter ?
Not that disturbing to me, but not very Javaish either :)

Reply:
I know, but getter/setter are so old-school...